src/java/org/apache/cassandra/io/compress/CompressionMetadata.java

```diff
@@ -263,35 +338,35 @@ public Chunk chunkFor(long position)
     public long getTotalSizeForSections(Collection<SSTableReader.PartitionPositionBounds> sections)
     {
         long size = 0;
-        long lastOffset = -1;
+        int lastIncludedChunkIdx = -1;
         for (SSTableReader.PartitionPositionBounds section : sections)
         {
-            int startIndex = (int) (section.lowerPosition / parameters.chunkLength());
-
-            int endIndex = (int) (section.upperPosition / parameters.chunkLength());
-            if (section.upperPosition % parameters.chunkLength() == 0)
-                endIndex--;
+            int sectionStartIdx = Math.max(chunkIndex(section.lowerPosition - uncompressedOffset), lastIncludedChunkIdx + 1);
```

Comment:
We are subtracting `uncompressedOffset` everywhere `chunkIndex` is used (except in the constructor where using `chunkIndex` feels wrong to me as the chunks have not been set up yet). Why not include it in the chunk index calculation, which will also make sure we lose the cognitive burden of thinking what kind of position to give it?